wordpress-mobile/WordPress-Android
# Polishing plugins

Fixes #6920. This PR wraps up the plugins feature except for analytics. A list of the final changes can be found in the issue. I've tried my best to explain the more complicated parts with comments, please let me know if any of them are not clear or if you have any suggestions to improve them.

I've taken a picture of the changes to the ` WordPress/src/main/res/layout/plugin_detail_activity.xml` file before wrapping it all up inside a `ScrollView` to make it easier for the reviewer. You can find that at the bottom of this PR description.

Since this is the last PR for plugins, we need to test pretty much everything as we'll merge the feature to `develop` once this PR is merged in.

To test:
* Pretty much everything about plugins needs to be tested
* We need to run these tests on several devices - different resolutions, api versions etc
* We need to make sure the feature works correctly with rotation changes
* A Jetpack site with at least 5.6 version installed is required
* I've spent quite a bit of time trying to make sure that UI is responsive at all times. Actions are queued up and hopefully never gets lost even if the user leaves the page. We should try to quickly change the state of the plugin (active & autoupdates) consecutively and maybe try to remove it at the same time and compare the result with what we'd have expected as a user.

These are the remarks I can think of right now, but I'll add it as a comment if I can think of anything else.

@nbradbury Do you mind taking this one on Monday when you get back? @kwonye said he could make sure that you're clear on the media project on Monday so you can focus on this one and I think we'll all feel a lot more comfortable with the feature if you did the last review :)

![plugin detail layout changes](https://user-images.githubusercontent.com/662023/33787663-584b92d6-dc3c-11e7-847c-10828c38e390.png)
